### PR TITLE
Add basic GitHub CI workflow

### PR DESCRIPTION
**Pull request by atoomic**
In addition to Travis CI run integrations
tests on GitHub using Perl 5.6 to 5.32
and a basic test on macOS and Windows.

Note that a fast a simple test is run using
a stock ubuntu docker image from GitHub.
Other jobs are only triggered on success.

### CODE REVIEW

**Comment by shogo82148** [comment]
FYI
Currently, CI of Scalar-List-Utils is running on travis-ci.org https://travis-ci.org/github/Dual-Life/Scalar-List-Utils
But Travis-ci.org shut down will occur on May 31, 2021. (ref. https://blog.travis-ci.com/2021-05-07-orgshutdown)
So we should migrate to travis-ci.com or another CI service.

**Comment by atoomic** [comment]
@haarg let me test and adjust it, and add at the same time the last versions of Perl up to 36 :-)

**Comment by atoomic** [comment]
@haarg I've adjusted the workflow, you can view a run from my repo here using changes from that PR.
https://github.com/atoomic/Scalar-List-Utils/actions/runs/2489469736

thanks

File: .github/workflows/testsuite.yml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  windows:
+    needs: [ubuntu, linux]
+    runs-on: windows-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Perl
+        run: |
+          choco install strawberryperl
+          echo 'PATH=C:\strawberry\c\bin;C:\strawberry\perl\site\bin;C:\strawberry\perl\bin' >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (atoomic) [reply .github/workflows/testsuite.yml:121]: worth giving it a try, let me check

[the remark this answers is not in the document]
Comment (atoomic) [reply .github/workflows/testsuite.yml]: now using GITHUB_PATH as part of cedc3da9 changes

**Comment by haarg** [comment]
Thanks!

**Pull request merged**